parser/frontmatter.go

```diff
@@ -174,17 +178,51 @@ func HandleTOMLMetaData(datum []byte) (interface{}, error) {
 	return m, nil
 }
 
+// removeTOMLIdentifier removes, if necessary, beginning and ending TOML
+// frontmatter delimiters from a byte slice.
 func removeTOMLIdentifier(datum []byte) []byte {
-	return bytes.Replace(datum, []byte(TOMLDelim), []byte(""), -1)
+	if len(datum) < 8 {
+		return datum
+	}
+
+	b := bytes.TrimPrefix(datum, []byte(TOMLDelim))
+	if len(datum)-len(b) == 3 {
```

Comment:
Something tells me that you can use bytes.TrimFunc or similar to handle this and below.